language/go/fix.go

```diff
@@ -171,15 +171,15 @@ func migrateLibraryEmbed(c *config.Config, f *rule.File) {
 	}
 }
 
-// migrateGrpcCompilers converts "go_grpc_library" rules into "go_proto_library"
-// rules with a "compilers" attribute.
+// migrateGrpcCompilers converts "go_proto_library" rules with a "compilers"
+// attribute containing exactly oldGrpcCompilerLabel into "go_grpc_library" rules.
 func migrateGrpcCompilers(c *config.Config, f *rule.File) {
 	for _, r := range f.Rules {
-		if r.Kind() != "go_grpc_library" || r.ShouldKeep() || r.Attr("compilers") != nil {
+		if r.Kind() != "go_proto_library" || r.ShouldKeep() || r.Attr("compilers") == nil || len(r.AttrStrings("compilers")) != 1 || !strListAttrContains(r, "compilers", oldGrpcCompilerLabel) {
```

Comment:
How can I check if the corresponding proto `.hasServices()` instead of checking for the old compiler name?

Reply:
I removed this fixer to unblock this PR as we do no longer need to convert in this direction. Existing rules will just be kept as is which should continue working. Adding a fixer to convert the opposite direction might be added in a future PR.